src/Engine/Audio/SteamAudioInterface.cpp

```diff
@@ -1,46 +1,128 @@
 #include "SteamAudioInterface.hpp"
+#include <assert.h>
 
-#ifdef USINGMEMTRACK
-#include <MemTrackInclude.hpp>
-#endif
+using namespace Audio;
 
 SteamAudioInterface::SteamAudioInterface() {
-    context = nullptr;
-    playerDir = new IPLVector3{ 0, 0, 0 };
-    playerPos = new IPLVector3{ 0, 0, 0 };
-    playerUp = new IPLVector3{ 0, 0, 0 };
+    simSettings.sceneType = IPL_SCENETYPE_PHONON;
+    simSettings.numRays = 12400;
+    simSettings.numDiffuseSamples = 1024;
+    simSettings.numBounces = 3;
+    simSettings.irDuration = 1.0;
+    simSettings.ambisonicsOrder = 1;
+    simSettings.maxConvolutionSources = 10;
+    context = IPLContext{ nullptr, nullptr, nullptr };
 }
 
 SteamAudioInterface::~SteamAudioInterface() {
-    delete context;
-    // Might have to remove these depending on how the pointers are sent to this class
-    delete playerDir;
-    delete playerPos;
-    delete playerUp;
+    if (environment)
+        iplDestroyEnvironment(&environment);
+    if (scene)
+        iplDestroyScene(&scene);
 }
 
-void SteamAudioInterface::Process(float* input, uint32_t samples,  IPLVector3 * sourcePosition, float sourceRadius) {
+void SteamAudioInterface::CreateScene(uint32_t numMaterials) {
+    if (scene) {
+        iplDestroyScene(&scene);
+        scene = NULL;
+    }
+    IPLerror err = iplCreateScene(context, NULL, simSettings, numMaterials, &scene);
+}
+
+void SteamAudioInterface::FinalizeScene(IPLFinalizeSceneProgressCallback progressCallback) {
+    if (scene)
+        iplFinalizeScene(scene, NULL);
+}
+
+void SteamAudioInterface::CreateRenderers(SteamAudioRenderers*& renderers) {
+    renderers = new SteamAudioRenderers(environment, sAudio.envRenderer, sAudio.binauralRenderer);
+}
+
+SteamAudioInterface::SaveData SteamAudioInterface::SaveFinalizedScene() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'SaveFinalizedScene' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)